pypqlx.py

```diff
@@ -73,19 +73,35 @@ def __init__(self, first, last, ndata, N, S, C, L = ''):
         self.L = L if L != "" else "--"
     
     def mode(self):
-        pass
-    
+        mode = []
+        for i in sorted(set(pdf.period)):
+            a = pdf.power[max(pdf.count[pdf.period == i])]
```

Comment:
You should have used the argmax doesn't ? I.e. you need to find the power where the histogram is larger for each period. Histogram amplitude is count. So, something like

```
pdf.power[pdf.period == i][np.argmax(pdf.count[pdf.period == i])]
```
 Needs to check!